main.js

```diff
@@ -0,0 +1,140 @@
+'use strict'
+
+// ホーム画面のDOM
+const homeContainer = document.getElementById('home-container');
+const main = document.getElementById('main');
+const para = document.getElementById('para');
+const btn = document.getElementById('btn');
+
+// 問題画面のDOM
+const questionContainer = document.getElementById('question-container');
+const questionAnswerContainer = document.getElementById('question-answers-container');
+const quizAnswerBtn = document.querySelectorAll('#question-answers-container > button');
+const quizTitle = document.getElementById('quiz-title');
+const quizGenre = document.getElementById('quiz-genre');
+const quizDif = document.getElementById('quiz-dif');
+const quizQuestion = document.getElementById('question');
+const quizAnswer0 = document.getElementById('answer0');
+const quizAnswer1 = document.getElementById('answer1');
+const quizAnswer2 = document.getElementById('answer2');
+const quizAnswer3 = document.getElementById('answer3');
+
+// 結果画面のDOM
+const result = document.getElementById('result');
+const resultTitle = document.getElementById('result-title');
+const resultPara = document.getElementById('result-para');
+const resultBtn = document.getElementById('result-btn');
+
+// 正解数用のカウンター
+let correctCounter = 0;
+
+// 何問目か用のカウンター
+let quizCounter = 0;
+
+// クイズ要素
+let quizObj = [];
+
+// 開始ボタンを押下すると実行する非同期処理
+btn.addEventListener('click', () => {
+  main.innerText = '取得中';
+  para.innerText = '少々お待ちください';
+  btn.classList.add('transparent');
+  getQuizData();
```

Comment:
`getQuizData` は `async` なので呼び出す際は `await` を付けて呼び出してください。
```suggestion
  await getQuizData();
```

Reply:
コードレビュー、プルリクの指摘ありがとうございます！

このように修正いたしました。

https://github.com/chillout2san/work_js4/blob/d60c0d2affca023e1353072643398fe4a9fddb7c/main.js#L37-L43

ちなみに非同期処理があまり理解できていないなと感じ、色々実験しておりました。
await getQuizData();を39行目に置き、「開始」ボタンを押下すると、「取得中」の画面に切り替わらず、そのままクイズ出題されました。

こちらはgetQuizData();が大して重い処理ではなかったために、「**取得中」の画面に切り替わっているが、見えなかった**という理解でよいでしょうか。

宜しくお願いいたします。